### PR TITLE
Windows: Fix devices enumeration

### PR DESCRIPTION
**Pull request by jm4R**
Based on
https://github.com/microsoft/Windows-driver-samples/blob/master/hid/hclient/pnp.c#L32

Fixes wrong detection of the class of devices.

Fixes: #234

### CODE REVIEW

File: windows/hid.c [hunk 1]
```diff
@@ -378,32 +381,18 @@ struct hid_device_info HID_API_EXPORT * HID_API_CALL hid_enumerate(unsigned shor
...
-				(strcmp(driver_name, "Keyboard") == 0)) {
-				/* See if there's a driver bound. */
-				res = SetupDiGetDeviceRegistryPropertyA(device_info_set, &devinfo_data,
-				           SPDRP_DRIVER, NULL, (PBYTE)driver_name, sizeof(driver_name), NULL);
-				if (res)
-					break;
-			}
-		}
+		/* Populate devinfo_data. This function will return failure
+		   when there are no more interfaces left. */
```
Comment (Youw) [inline windows/hid.c:385]: please update or remove this comment, it is outdated

File: windows/hid.c [hunk 2]
```diff
@@ -309,12 +309,15 @@ struct hid_device_info HID_API_EXPORT * HID_API_CALL hid_enumerate(unsigned shor
...
 	GUID InterfaceClassGuid = {0x4d1e55b2, 0xf16f, 0x11cf, {0x88, 0xcb, 0x00, 0x11, 0x11, 0x00, 0x00, 0x30} };
+
+	/* Windows objects for interacting with the driver. */
 	SP_DEVINFO_DATA devinfo_data;
 	SP_DEVICE_INTERFACE_DATA device_interface_data;
 	SP_DEVICE_INTERFACE_DETAIL_DATA_A *device_interface_detail_data = NULL;
 	HDEVINFO device_info_set = INVALID_HANDLE_VALUE;
+	char driver_name[256];
 	int device_index = 0;
 	int i;
```
Comment (Youw) [inline windows/hid.c:322]: remove it completely?

**Review by Youw — APPROVED** [review]
what are the chances we will fail to filter out a non-hid devices?
I understand that HID class GUID should do the trick, but still, this implementation been here like forever, and I'like to give it some thinking, just in case.

File: windows/hid.c (hunk 2 above)
Comment (jm4R) [reply windows/hid.c:322]: What about "Make sure this device has a driver bound to it" section? This buffer is used there. I am not sure why we check it - HID device without any driver should still work, right?

Comment (Youw) [reply windows/hid.c]: yeap, only `i`

**Comment by jm4R** [comment]
> what are the chances we will fail to filter out a non-hid devices?
> I understand that HID class GUID should do the trick, but still, this implementation been here like forever, and I'like to give it some thinking, just in case.

I am not WinApi expert but because this PR is based on the code straight from Microsoft, I assume it should be correct. I also read the documentation of each changed function and indicated that it should work. If the bug I fixed by this PR was there "forever" too (and I am 100% sure it is a bug), I am confused nobody discovered it yet. But maybe some people did - there are a number of open issues in the old repository, that could be caused by this. For example:
https://github.com/signal11/hidapi/issues/290
https://github.com/signal11/hidapi/issues/221

In my use case the old code didn't work (described in #234), but it is my use case :)

**Pull request merged**